### DISCUSSION TITLE
Is there a way to allow subscriptions for multiple user accounts (on own backend) using the same apple ID?

### DISCUSSION THREAD

**Discussion by TheFrostage** [hyochan/react-native-iap#1470] [Q&A]
Hello everyone! I've been struggling to find a way to handle this one use case of my application. 
I want to allow users to purchase auto-renewable subscriptions linked to their user profiles handled by my backend server. 

This is the flow that reproduces the use case:
1. Register a new user profile or sign into a profile without active subscription.
2. "Choose Subscription" screen is displayed, where user purchases an auto-renewable annual subscription.
3. After successful purchase, backend validates the purchase and unlocks full app functionality. App navigates to main screen.
4. User logs out of the app and creates another profile/sign into a profile without active subscription.
5. "Choose Subscription" screen is displayed ONCE AGAIN, where user CANNOT purchase the same type of subscription because he already has one associated with current Apple ID. 

I understand that my backend server should be the source of truth about subscriptions and stuff. Where I'm struggling is on step 5, and I don't understand how to either allow users to purchase multiple subscriptions or to "soft-lock" them.

**Comment by darimuhittinhey** [comment]
Same issue here. With this scheme our backend cant be the source of truth. Anyone can help ?

**Comment by a-klotz-p8** [comment]
Facing the same issue. Anyone have any tips?
